source/api/discovery/0.1/index.md

```diff
@@ -0,0 +1,633 @@
+  	"type": "OrderedCollection"
+  },
+  "prev": {
+  	"id": "https://example.org/activity/page-0",
+  	"type": "OrderedCollectionPage"
+  },
+  "next": {
+  	"id": "https://example.org/activity/page-2",
+  	"type": "OrderedCollectionPage"
+  },
+  "orderedItems": [
+     {
+     	"type": "Update",
+     	"object": {
+     		"id": "https://example.org/iiif/1/manifest",
+     		"type": "Manifest"
+     	},
+     	"endTime": "2018-03-10T10:00:00Z"
+     }
+  ]
+}
+```
+
+### 3.3. Activities
+
+The Activities are the means of describing the changes that have occured in the content provider's system.
+
+Content providers _MAY_ publish Activities separately from Ordered Collection Pages, and if so they _MUST_ be at the HTTP(S) URI given in the `id` property of the Activity.
+
+##### id
+
+An identifier for the Activity.
+
+Activities _MAY_ have an `id` property. The value _MUST_ be a string and it _MUST_ be an HTTP(S) URI. The JSON representation of the Activity _MAY_ be available at the URI.
+
+```
+{ "id": "https://example.org/activity/1" }
+```
+
+##### type
+
+The type of Activity. 
+
+This specification uses the types described in the table below.
+
+| Type   | Definition |
+| ------ | ---------- |
+| `Create` | The initial creation of the resource.  Each resource _MUST_ have at most one `Create` Activity in which it is the `object`. |
+| `Update` | Any change to the resource.  In a system that does not distinguish creation from modification, then all changes _MAY_ have the `Update` type. |
+| `Delete` | The deletion of the resource, or its de-publication from the web. Each resource _MUST_ have at most one Activity in which it is the `object`. |
+{: .api-table #table-type-dfn}
+
+Activities _MUST_ have the `type` property. The value _MUST_ be a registered Activity class, and _SHOULD_ be one of `Create`, `Update`, or `Delete`.
+
+```
+{ "type": "Update" }
+```
+
+##### object
+
+The IIIF resource that was affected by the Activity.  It is an implementation decision whether there are separate lists of Activities, one per object type, or a single list with all of the object types combined.
+
+Activities _MUST_ have the `object` property.  The value _MUST_ be a JSON object, with the `id` and `type` properties.  The `id` _MUST_ be an HTTP(S) URI. The `type` _MUST_ be a class defined in the IIIF Presentation API, and _SHOULD_ be one of `Collection`, or `Manifest`.
+
+```
+{
+  "object": {
+  	"id": "http://example.org/iiif/1/manifest",
+  	"type": "Manifest"
+  }
+}
+```
+
+##### endTime
+
+The time at which the Activity was finished. It is up to the implementer to decide whether the Activity includes the publication of the IIIF resource online, or only the internal data modification, but the decision _MUST_ be consistently applied. 
+
+Activities _SHOULD_ have the `endTime` property.  The value _MUST_ be a datetime expressed in UTC in the [xsd:dateTime][org-w3c-xsd-datetime] format.
+
+```
+{ "endTime": "2017-09-21T00:00:00Z" }
```

Comment:
Can an endTime be in the future?

Reply:
(almost certainly not, but...)

Reply:
I think there's a hypothetical use case where it is -- you're alerting that some update /will/ happen, rather than has already happened.  Not sure we need to specify either way?

Reply:
The first comment is the bigger issue. I publish my updates, you come along and harvest them, notice that resource X has changed since you were last here, and fetch the resource from the CDN. The change has been made in my internal system, but the cache hasn't expired yet at the CDN. Yes, I know I could mitigate against this, but the spec should say that I need to ensure the resource is really ready for reindexing at its public URL.

Reply:
I think the important point is that the changed resource must be available at the URI before the time listed in the activity.  If it's in the future, then it's a promise. If it's in the past, then it must already be available.

Reply:
I agree with the above, the expectation is that the "resource" (which is web speak means the thing at URI that you can GET a representation of -- irrespective of whether served by CDN or other means) should be available by `endTime`

Reply:
I understand but was really thinking about future ‘endTimes’ - my hypothetical case was that an implementation behind a CDN might benefit from publishing ’endTime’s in the future when the resource would be universally available in its updated form - for example, a simple implementation that’s serializing the Update entries as the new Manifests are written.   It’s easy enough to work around, but was something I hadn’t considered until I read @tomcrane ‘s comment.

Reply:
I am OK with the new version. I just have the observation that it puts more burden on the consumer, when chunking through new activities with endTimes later than the previous crawl, it has to be aware that the endTime might be in the future and it can't go and get that resource right now - it might have to shelve that reindex, queue it somewhere etc - which requires a more sophisticated harvester capable of deferring fetches for non-trivial amounts of time.